### PR TITLE
Added Domain Resolution

### PR DESCRIPTION
**Pull request by Isaac0-dev**

### CODE REVIEW

**Review by Beyley — APPROVED** [review]
Tested on Arch Linux, works perfectly

File: src/pc/djui/djui_panel_join.c [hunk 1]
```diff
@@ -87,12 +88,12 @@ static bool djui_panel_join_ip_valid(char* buffer) {
 }
 
 static void djui_panel_join_ip_text_change(struct DjuiBase* caller) {
```
Comment (djoslin0) [inline src/pc/djui/djui_panel_join.c]: remove function if it's unused now

File: src/pc/network/socket/domain_res.c [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  char *host_name;
+  struct hostent *remoteHost;
+  char* domainname = "";
+  host_name = configJoinIp;
+  if (host_name == NULL) {
+	  return;
+  }
+  int i = 0;
+  remoteHost = gethostbyname(host_name);
+  i = 0;
```
Comment (djoslin0) [inline src/pc/network/socket/domain_res.c:25]: this line is unnecessary

File: src/pc/network/socket/domain_res.c [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  struct in_addr addr;
+  char *host_name;
+  struct hostent *remoteHost;
+  char* domainname = "";
+  host_name = configJoinIp;
+  if (host_name == NULL) {
+	  return;
+  }
+  int i = 0;
+  remoteHost = gethostbyname(host_name);
```
Comment (djoslin0) [inline src/pc/network/socket/domain_res.c:24]: remove host_name variable, just pass `configJoinIp` directly into `gethostbyname()`

Comment (djoslin0) [reply src/pc/network/socket/domain_res.c:24]: keep the NULL check, but change it to `configJoinIp`

File: src/pc/djui/djui_panel_join.c [hunk 4]
```diff
@@ -136,11 +128,11 @@ static void djui_panel_join_ip_text_set(struct DjuiInputbox* inputbox1) {
 }
 
 void djui_panel_join_do_join(struct DjuiBase* caller) {
-    if (!djui_panel_join_ip_valid(sInputboxIp->buffer)) {
+    if (!(strlen(sInputboxIp->buffer) > 0)) { 
         djui_interactable_set_input_focus(&sInputboxIp->base);
         djui_inputbox_select_all(sInputboxIp);
         return;
-    }
+        }
```
Comment (djoslin0) [inline src/pc/djui/djui_panel_join.c]: decrease indentation of `}`

**Pull request merged**